instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java

```diff
@@ -84,7 +85,10 @@ public Resource createResource(ConfigProperties config) {
             this::findByCurrentDirectoryApplicationYaml,
             this::findByClasspathApplicationProperties,
             this::findByClasspathApplicationYml,
-            this::findByClasspathApplicationYaml);
+            this::findByClasspathApplicationYaml,
+            this::findByClasspathBootstrapProperties,
```

Comment:
@AlchemyDing did you verify that these have lower precedence than the source that are listed before them?

Reply:
Yes, when spring.application.name appears in both application.yml and bootstrap.yml, spring will take it from application.yml.

Reply:
in application.properties,spring.application.name=otel-demo-application;
in bootstrap.properties,spring.application.name=otel-demo-bootstrap-prop
environment.getProperty("spring.application.name")==otel-demo-application
![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/88769332/e9875484-1dde-4aaa-b042-b4d07563b5a1)

Reply:
[https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies](url)

Reply:
@laurit 
[https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies](url)

Reply:
[https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies](https://docs.spring.io/spring-cloud-commons/docs/4.0.4/reference/html/#application-context-hierarchies)
 @laurit

Reply:
Thanks for verifying this